GetSunSetRise.cpp

```diff
@@ -114,70 +112,77 @@ void GetSunSetRise::clickedCity(QAction *pAction)
 
 qreal GetSunSetRise::zoneToTime(const int &index)
 {
-    if(timeZones.size() < index + 1)
-    {
-        QMessageBox::critical(0, "ERROR", "The time zone was selected an incorrect way!");
-        return 0;
+    if(m_timeZones.size() < index + 1) {
+        QMessageBox::critical(nullptr, "ERROR", "The time zone was selected an incorrect way!");
+
+        return 0.0;
     }
-    return timeZones.at(index);
+
+    return m_timeZones.at(index);
 }
 
-QString GetSunSetRise::toHumanTime(float preResult)
+QString GetSunSetRise::toHumanTime(double preResult)
 {
-    QString hours = QString::number((int)preResult);
-    QString minuts = QString::number(BLib::round(preResult - (int)preResult, 2));
+    QString hours = QString::number(static_cast<int>(preResult));
+    QString minutes = QString::number(functions::round(preResult - static_cast<double>(static_cast<int>(preResult)), 2));
 
-    switch (hours.size())
-    {
-    case 0: hours.push_front('0');
+    switch (hours.size()) {
+    case 0:
+        hours.push_front('0');
         hours.push_front('0');
         break;
-    case 1: hours.push_front('0');
+    case 1:
+        hours.push_front('0');
+        break;
+    default:
         break;
-    default: break;
     }
 #ifdef DEBUG
     qDebug() << "minuts.size() =" << minuts.size();
 #endif
-    switch (minuts.size())
-    {
-    case 0: minuts.push_back('0');
-        minuts.push_back('0');
+    // maximum size of minutes if 5: "-0.xx". It is always less than one in absolute value
+    switch (minutes.size()) {
+    case 0: // ""
+        minutes.push_back('0');
+        minutes.push_back('0');
         break;
-    case 1: minuts.push_back('0');
+    case 1: // "0"
+        minutes.push_back('0');
         break;
-    case 3: minuts.remove(0, 2);
-        minuts.push_back('0');
+    case 3: // "0.x"
+        minutes.remove(0, 2); // remove "0."
+        minutes.push_back('0');
         break;
-    case 4: if(minuts < 0)
-        {
-            minuts.remove(0, 3);
-            minuts.push_back('0');
-        }
-        else
-        {
-            minuts.remove(0, 2);
+    case 4:
+        if(minutes.indexOf("-") != -1) { // "-0.x"
+            minutes.remove(0, 3);
+            minutes.push_back('0');
+        } else { // "0.xx"
+            minutes.remove(0, 2);
         }
         break;
-    case 5: minuts.remove(0, 3);
+    case 5:
+        minutes.remove(0, 3); // remove "-0."
         break;
-    default: minuts.remove(0, 2);
+    default: // "0."
+        minutes.remove(0, 2);
         break;
     }
 #ifdef DEBUG
     qDebug() << "minuts.size() =" << minuts.size();
```

Comment:
This is not actual code.
```suggestion
    qDebug() << "minutes.size() =" << minutes.size();
```

Reply:
This code no longer exist.